src/models/result.ts

```diff
@@ -0,0 +1,127 @@
+import { Company } from './company.js';
+import { Position } from './positions.js';
+import { Provider } from './provider.js';
+
+/**
+ * An enriched version of a Segment that includes full position objects instead of just IDs.
+ */
+export interface EnrichedSegment {
+	id: string;
+	departurePosition: Position;
+	arrivalPosition: Position;
+	departureTime: string;
+	arrivalTime: string;
+	duration?: string;
+}
+
+/**
+ * A stop (connection) detail between segments. For every connection we list
+ * the arrival position of the previous segment, the departure position of the next,
+ * and a string indicating the connecting time.
+ */
+export interface StopDetail {
+	departure: string;
+	arrival: string;
+	departureTime: string;
+	arrivalTime: string;
+	duration: string;
+	waitTime?: string;
+}
+
+export interface Result {
+	company: Company;
+	departurePosition: Position;
+	arrivalPosition: Position;
```

Comment:
omio uses departure/arrival station so I'd use that to keep it consistent

![image](https://github.com/user-attachments/assets/85153647-3f73-4d1f-bbb6-e80e504725ff)

![image](https://github.com/user-attachments/assets/8ba9be2d-f3d8-43ae-aebe-87fa920fbd79)